app/actions/index.ts

```diff
@@ -0,0 +1,31 @@
+import {PopUpState} from '../types'
+//ACTION CONSTANTS
+export const  START_LISTENING = "START_LISTENING";
+export const STOP_LISTENING = "STOP_LISTENING";
+export const  ERROR = "ERROR";
+export const  DEFAULT_ACTION = "DEFAULT";
+export const CLEAR_REQUESTS = "CLEAR_REQUESTS"
+export const RELOAD_REQUESTS = "RELOAD_REQUESTS"
+export const  UPDATE_FIELD = "UPDATE_FIELD";
+export const  UPDATE_FIELDS = "UPDATE_FIELDS";
+
+// Action Creators
+
+export function startListening (enabledStatus:PopUpState["enabled"]){
```

Comment:
If you find yourself trying to access a type's member to use that member's type, it's a good idea to alias that type and use that instead. This could be a type of `PopUpStatus`, and use that both in `PopUpState` here.